src/test/librbd/test_Groups.cc

```diff
@@ -749,3 +752,84 @@ TEST_F(TestGroup, snap_list2PP)
   ASSERT_EQ(0, m_rbd.group_remove(m_ioctx, gp_name));
   ASSERT_EQ(0, _rados.pool_delete(pool_name2.c_str()));
 }
+
+TEST_F(TestGroup, snap_list_internal)
+{
+  REQUIRE_FORMAT_V2();
+
+  // Check that the listing works with different
+  // values for try_to_sort and fail_if_not_sorted
+
+  librados::IoCtx ioctx;
+  ASSERT_EQ(0, _rados.ioctx_create(_pool_name.c_str(), ioctx));
+
+  const char *group_name = "gp_snaplist_internalPP";
+
+  librbd::RBD rbd;
+  ASSERT_EQ(0, rbd.group_create(ioctx, group_name));
```

Comment:
What about
>  * a test where a stale order key for a snapshot that isn't there is encountered, showing that that doesn't affect the behavior, irrespective of `try_to_sort` or `fail_if_not_ordered` values

It seems like it would be sufficient to create a dummy snapshot, get its ID and remove its `snap` key from RADOS here.  The remainder of the test would then cover this scenario with no changes.